client/app/visualizations/chart/Editor/XAxisSettings.jsx

```diff
@@ -1,9 +1,15 @@
+import { includes, isFinite, isNumber } from 'lodash';
 import React from 'react';
-import { Section, Switch } from '@/components/visualizations/editor';
+import { Section, InputNumber, Switch } from '@/components/visualizations/editor';
 import { EditorPropTypes } from '@/visualizations';
 
 import AxisSettings from './AxisSettings';
 
+function toNumber(value) {
+  value = isNumber(value) ? value : parseFloat(value);
+  return isFinite(value) ? value : null;
+}
+
```

Comment:
Consider using existing utility for this: https://github.com/getredash/redash/blob/a1255b41442da27b8bff70f70018265b7a2a1540/viz-lib/src/visualizations/chart/plotly/utils.js#L5-L7